x-pack/plugin/async-search/src/main/java/org/elasticsearch/xpack/search/AsyncSearchTask.java

```diff
@@ -437,10 +455,14 @@ protected void onListShards(
                 delegate = new CCSSingleCoordinatorSearchProgressListener();
                 delegate.onListShards(shards, skipped, clusters, fetchPhase, timeProvider);
             }
-            searchResponse.compareAndSet(
-                null,
-                new MutableSearchResponse(shards.size() + skipped.size(), skipped.size(), clusters, threadPool.getThreadContext())
+            MutableSearchResponse mutableSearchResponse = new MutableSearchResponse(
+                shards.size() + skipped.size(),
+                skipped.size(),
+                clusters,
+                threadPool.getThreadContext()
             );
+            searchResponse.compareAndSet(null, mutableSearchResponse);
```

Comment:
Just to highlight the actual changes we are making, would it make sense to restore this as it was?

Reply:
Yes. Restored in next push.

Reply:
Yes, fixed.